public/components/common/search/direct_search.tsx

```diff
@@ -258,6 +296,16 @@ export const DirectSearch = (props: any) => {
 
     if (status === DirectQueryLoadingStatus.SUCCESS || datarows) {
```

Comment:
will this `|| datarows` always be true?

Reply:
I didn't write this code originally, but I believe it won't be, as the only way that the datarows field will be inside of the polling result is if results are returned successfully.